File: template/src/main/java/ru/surfstudio/standard/ui/base/configurator/ActivityConfigurator.java
package ru.surfstudio.standard.ui.base.configurator;

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator;
import ru.surfstudio.standard.app.App;
import ru.surfstudio.standard.app.dagger.AppComponent;
import ru.surfstudio.standard.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.standard.ui.base.dagger.activity.ActivityModule;
import ru.surfstudio.standard.ui.base.dagger.activity.DaggerActivityComponent;

/**
 * Базовый конфигуратор для активити
 */

public class ActivityConfigurator extends BaseActivityConfigurator<ActivityComponent, AppComponent> {

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(new ActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) getTargetActivity().getApplicationContext()).getAppComponent();
    }
}

File: template/src/main/java/ru/surfstudio/standard/ui/base/configurator/ActivityConfigurator.kt
package ru.surfstudio.standard.ui.base.configurator

import android.content.Context

import ru.surfstudio.android.core.ui.base.screen.configurator.BaseActivityConfigurator
import ru.surfstudio.standard.app.App
import ru.surfstudio.standard.app.dagger.ActivityComponent
import ru.surfstudio.standard.app.dagger.AppComponent
import ru.surfstudio.standard.app.dagger.DaggerActivityComponent


class ActivityConfigurator(private val context: Context) : BaseActivityConfigurator<ActivityComponent, AppComponent>() {

    override fun createActivityComponent(parentComponent: AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .build()
    }

    override fun getParentComponent(): AppComponent? {
        return (context.applicationContext as App).appComponent
    }
}
